src/ucp/core/ucp_ep.c

```diff
@@ -1143,6 +1143,36 @@ static void ucp_ep_set_lanes_failed(ucp_ep_h ep, uct_ep_h *uct_eps)
     }
 }
 
+static int ucp_can_uct_ep_unprogress(ucp_ep_h ep, uct_ep_h uct_ep)
+{
+    if (!ep->worker->context->config.ext.adaptive_progress ||
+        ucp_is_uct_ep_failed(uct_ep)) {
+        return 0;
+    }
+
+    if (!ucp_wireup_ep_test(uct_ep) ||
+        ((ucp_wireup_ep(uct_ep)->super.uct_ep != NULL) &&
+        ucp_wireup_ep(uct_ep)->super.is_owner)) {
+        return 1;
+    }
+
+    return 0;
+}
+
+void ucp_ep_unprogress(ucp_ep_h ep, uct_ep_h uct_ep,
+                       ucp_rsc_index_t rsc_index)
```

Comment:
pass `lane` as an argument and get `rsc_index` by `ucp_ep_get_rsc_index(ep, lane)` inside of this func

Reply:
i also thought it at first, but when we discard from wireup_ep there is no lane:
https://github.com/openucx/ucx/pull/7819/files#diff-5c20ab9777a62538100bec00f0ce2b19db87f7cd5c2553d15a62438fd465254fR334
also, it can be safer to use rsc_index in case of ep configuration change after discard

Reply:
ok